libs/sdk-ui-theme-provider/src/derivedColors.ts

```diff
@@ -0,0 +1,217 @@
+    getCssProperty("palette-error-base-d20", palette?.error?.base && darken(0.2, palette.error.base)),
+
+    getCssProperty(
+        "palette-success-disabled",
+        palette?.success?.base && transparentize(0.5, lighten(0.06, palette.success.base)),
+    ),
+    getCssProperty(
+        "palette-success-focus",
+        palette?.success?.base && transparentize(0.5, lighten(0.06, palette.success.base)),
+    ),
+    getCssProperty("palette-success-base-d06", palette?.success?.base && darken(0.06, palette.success.base)),
+    getCssProperty("palette-success-base-d12", palette?.success?.base && darken(0.12, palette.success.base)),
+];
+
+const getBubbleDerivedColors = (palette: IThemePalette): CssProperty[] => [
+    getCssProperty(
+        "palette-primary-base-t10",
+        palette?.primary?.base && transparentize(0.1, palette.primary.base),
+    ),
+    getCssProperty("palette-error-base-t10", palette?.error?.base && transparentize(0.1, palette.error.base)),
+];
+
+const getDateFilterDerivedColors = (palette: IThemePalette): CssProperty[] => [
+    getCssProperty(
+        "palette-primary-base-lighten45",
+        palette?.primary?.base && lighten(0.45, palette.primary.base),
+    ),
+];
+
+const getMeasureNumberFormatDialogDerivedColors = (palette: IThemePalette): CssProperty[] => [
+    getCssProperty(
+        "palette-primary-base-darken20",
+        palette?.primary?.base && darken(0.2, palette.primary.base),
+    ),
+];
+
+const getPivotTableDerivedColors = (palette: IThemePalette): CssProperty[] => [
+    getCssProperty(
+        "palette-primary-base-dimmed-darken03",
+        palette?.primary?.base && darken(0.03, mix(0.1, palette.primary.base, GD_COLOR_TEXT_LIGHT)),
+    ),
+];
+
+const getFormDerivedColors = (palette: IThemePalette): CssProperty[] => [
+    getCssProperty("palette-error-base-t50", palette?.error?.base && transparentize(0.5, palette.error.base)),
+    getCssProperty(
+        "palette-error-base-t70d20",
+        palette?.error?.base && transparentize(0.7, darken(0.2, palette.error.base)),
+    ),
+    getCssProperty(
+        "palette-warning-base-t70d20",
+        palette?.warning?.base && transparentize(0.7, darken(0.2, palette.warning.base)),
+    ),
+];
+
+/**
+ * @internal
+ */
+export const generateDerivedColors = (palette: IThemePalette): CssProperty[] =>
+    (palette &&
+        [
+            ...getCommonDerivedColors(palette),
+            ...getMessagesDerivedColors(palette),
+            ...getDashboardsDerivedColors(palette),
+            ...getButtonDerivedColors(palette),
+            ...getBubbleDerivedColors(palette),
+            ...getDateFilterDerivedColors(palette),
+            ...getMeasureNumberFormatDialogDerivedColors(palette),
+            ...getPivotTableDerivedColors(palette),
+            ...getFormDerivedColors(palette),
+        ].filter((property) => !!property)) ||
+    [];
```

Comment:
small stuff: this function is too long to be using the shortened syntax (with braces and return).